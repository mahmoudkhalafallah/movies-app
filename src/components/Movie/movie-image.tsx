import React, { useState } from "react";
import styled from "styled-components/native";
import { IMAGE_URI_PREFIX } from "../../utils/Constants";

const imagePlaceholder = require('../../../assets/images/film-poster-placeholder.png')

const StyledImage = styled.Image`
width: ${(props) => props.horizontal? '100' : '150'};
height: ${(props) => props.horizontal? '100' : '200'};
margin-right: ${(props) => props.horizontal? '10' : '20'};
border-top-left-radius: 10;
border-top-right-radius: 10;
border-bottom-left-radius: 10;
border-bottom-right-radius: 10;
`

export const MovieImage: React.FC<{ horizontal: string, src?: string }> = ({ horizontal, src }) => {

    const [err, setErr] = useState(false)
    const source = (!err && src) ? { uri: `${IMAGE_URI_PREFIX}${src}` } : imagePlaceholder

    return <StyledImage
        source={source}
        horizontal={horizontal}
        onError={() => setErr(true)}
    />
}