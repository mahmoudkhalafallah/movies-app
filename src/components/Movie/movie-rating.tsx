import React from "react";
import SwipeableRating from 'react-native-swipeable-rating';
import { Variables } from "../../utils/Variables";
import styled from "styled-components/native";
import { StyledText } from "../Text";


const RatingContainer = styled.View`
flex-direction: row;
align-items: center;
`

const RatingValue = styled(StyledText)`
font-size: 10;
color: #bbb;
`

export const MovieRating: React.FC<{ rating: number, horizontal: boolean }> = ({ rating, horizontal }) => {

    const ratingPerFiveStars = (rating / 10) * 5
    return <RatingContainer>
        <SwipeableRating
        color={Variables.ratingStarColor}
        emptyColor={Variables.ratingStarColor}
        rating={ratingPerFiveStars}
        size={15}
        gap={0.1}
        xOffset={20}
        allowHalves={true}
        swipeable={false}
    />
        {(!horizontal) && <RatingValue>{rating}/10</RatingValue>}
    </RatingContainer>
}
