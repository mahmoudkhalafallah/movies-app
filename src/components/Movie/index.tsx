import React from "react";

import { MovieTitle } from "../Titles";
import { MovieImage } from "./movie-image";
import { MovieRating } from "./movie-rating";
import styled from "styled-components/native";
import { StyledText } from "../Text";
import { IMAGE_URI_PREFIX } from "../../utils/Constants";

const MovieContainer = styled.View`
flex-direction: ${({ isMyMovie }: any) => isMyMovie ? 'column' : 'row'};
justify-content: space-between;
align-items: ${({ isMyMovie }: any) => isMyMovie ? 'center' : 'flex-start'};
width: 100;
margin-top: 10;
margin-bottom: 10;
${({ isMyMovie, isFirst }: any) => (isMyMovie && isFirst) ? 'margin-left: 25' : 'margin-left: 10'}
${({ isMyMovie, isLast }: any) => (isMyMovie && isLast) ? 'margin-right: 20' : 'margin-right: 10'}
${({ isMyMovie }: any) => (!isMyMovie) && 'margin-left: 20'}
${({ isMyMovie, isLast }: any) => (!isMyMovie && isLast) && 'padding-bottom: 80'}
`

const MovieDetails = styled.View`
flex-direction: column;
`

const MovieOverview = styled(StyledText)`
font-size: 13;
font-family: 'Oswald-Light';
margin-top: 5;
`

const MovieDate = styled(StyledText)`
font-size: 10;
margin-top: 5;
`

export const Movie: React.FC<any> = ({ title, poster_path, vote_average, overview,
    release_date, isFirst, isLast, isMyMovie }: any) => {
    return <MovieContainer isFirst={isFirst} isLast={isLast} isMyMovie={isMyMovie}>
        <MovieImage src={isMyMovie ? poster_path : `${IMAGE_URI_PREFIX}${poster_path}`} isMyMovie={isMyMovie}></MovieImage>
        <MovieDetails style={{width: isMyMovie ? 100 : 150}}>
            <MovieTitle ellipsizeMode='tail' numberOfLines={1}>{title}</MovieTitle>
            {(vote_average != undefined) && <MovieRating rating={vote_average} showRatingValue={isMyMovie} />}
            {!isMyMovie && <>
                <MovieDate>{release_date}</MovieDate>
                <MovieOverview ellipsizeMode='tail' numberOfLines={6}>{overview}</MovieOverview>
            </>}
        </MovieDetails>
    </MovieContainer>
}