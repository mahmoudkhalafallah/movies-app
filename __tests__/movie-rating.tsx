import 'react-native';
import React from 'react';
import { MovieRating } from '../src/components/Movie/movie-rating';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('rating renders correctly', () => {
  const rating = renderer.create(<MovieRating rating={5} />).toJSON();
  expect(rating).toMatchSnapshot();
});
