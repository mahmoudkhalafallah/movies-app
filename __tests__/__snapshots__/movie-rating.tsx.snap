// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rating renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-start",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 15,
            },
            Object {
              "color": "#e8b923",
              "marginRight": 0.1,
            },
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 15,
            },
            Object {
              "color": "#e8b923",
              "marginRight": 0.1,
            },
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 15,
            },
            Object {
              "color": "#e8b923",
              "marginRight": 0.1,
            },
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 15,
            },
            Object {
              "color": "#e8b923",
              "marginRight": 0.1,
            },
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 15,
            },
            Object {
              "color": "#e8b923",
              "marginRight": 0.1,
            },
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#bbb",
          "fontFamily": "Oswald-SemiBold",
          "fontSize": 10,
        },
      ]
    }
  >
    5
    /10
  </Text>
</View>
`;
